### ISSUE TITLE
Panic with deploying mpi-operator

### ISSUE THREAD

**Issue by jefflantz** [hashicorp/terraform-provider-kubernetes-alpha#91]
### Terraform Version and Provider Version
```
Terraform v0.12.28
+ provider.kubernetes-alpha (unversioned), compiled manually ~early July
```

### Kubernetes Version
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:52:00Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:43:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```
### Affected Resource(s)
```
kubernetes_manifest
 - crd
 - mpijob
```

### Terraform Configuration Files
The deployment is broken into four files due to dependencies. Each file is copied into the working directory and a terraform apply is run. These files come from [kubeflow/mpi-operator](https://github.com/kubeflow/mpi-operator/blob/master/deploy/v1/mpi-operator.yaml) and I used `tfk8s` to convert the resources to manifests then manually specified `provider = kubernetes-alpha`. Earlier I manually converted some of the resources to regular kubernetes resources (e.g., `kubernetes_cluster_role` instead of `kubernetes_manifest`), but that didn't seem to make a difference.
```hcl
provider "kubernetes-alpha" {
  config_path = "~/.kube/config"
  server_side_planning = true
}
resource "kubernetes_manifest" "namespace_mpi_operator" {
  provider = kubernetes-alpha
  manifest = {
    "apiVersion" = "v1"
    "kind" = "Namespace"
    "metadata" = {
      "name" = "mpi-operator"
    }
  }
}
```
```hcl
resource "kubernetes_manifest" "clusterrole_mpi_operator" {
  provider = kubernetes-alpha
  manifest = {
    "apiVersion" = "rbac.authorization.k8s.io/v1"
    "kind" = "ClusterRole"
    "metadata" = {
      "name" = "mpi-operator"
      "namespace" = "default"
    }
    "rules" = [
      {
        "apiGroups" = [
          "",
        ]
        "resources" = [
          "configmaps",
          "serviceaccounts",
        ]
        "verbs" = [
          "create",
          "list",
          "watch",
        ]
      },
      {
        "apiGroups" = [
          "",
        ]
        "resources" = [
          "pods",
        ]
        "verbs" = [
          "create",
          "get",
          "list",
          "watch",
          "delete",
          "update",
          "patch",
        ]
      },
      {
        "apiGroups" = [
          "",
        ]
        "resources" = [
          "pods/exec",
        ]
        "verbs" = [
          "create",
        ]
      },
      {
        "apiGroups" = [
          "",
        ]
        "resources" = [
          "endpoints",
        ]
        "verbs" = [
          "create",
          "get",
          "update",
        ]
      },
      {
        "apiGroups" = [
          "",
        ]
        "resources" = [
          "events",
        ]
        "verbs" = [
          "create",
          "patch",
        ]
      },
      {
        "apiGroups" = [
          "rbac.authorization.k8s.io",
        ]
        "resources" = [
          "roles",
          "rolebindings",
        ]
        "verbs" = [
          "create",
          "list",
          "watch",
        ]
      },
      {
        "apiGroups" = [
          "policy",
        ]
        "resources" = [
          "poddisruptionbudgets",
        ]
        "verbs" = [
          "create",
          "list",
          "update",
          "watch",
        ]
      },
      {
        "apiGroups" = [
          "apiextensions.k8s.io",
        ]
        "resources" = [
          "customresourcedefinitions",
        ]
        "verbs" = [
          "create",
          "get",
        ]
      },
      {
        "apiGroups" = [
          "kubeflow.org",
        ]
        "resources" = [
          "mpijobs",
          "mpijobs/finalizers",
          "mpijobs/status",
        ]
        "verbs" = [
          "*",
        ]
      },
      {
        "apiGroups" = [
          "scheduling.incubator.k8s.io",
          "scheduling.sigs.dev",
        ]
        "resources" = [
          "queues",
          "podgroups",
        ]
        "verbs" = [
          "*",
        ]
      },
    ]
  }
}

resource "kubernetes_manifest" "serviceaccount_mpi_operator" {
  provider = kubernetes-alpha
  manifest = {
    "apiVersion" = "v1"
    "kind" = "ServiceAccount"
    "metadata" = {
      "name" = "mpi-operator"
      "namespace" = "mpi-operator"
    }
  }
}

resource "kubernetes_manifest" "clusterrolebinding_mpi_operator" {
  provider = kubernetes-alpha
  manifest = {
    "apiVersion" = "rbac.authorization.k8s.io/v1"
    "kind" = "ClusterRoleBinding"
    "metadata" = {
      "name" = "mpi-operator"
      "namespace" = "mpi-operator"
    }
    "roleRef" = {
      "apiGroup" = "rbac.authorization.k8s.io"
      "kind" = "ClusterRole"
      "name" = "mpi-operator"
    }
    "subjects" = [
      {
        "kind" = "ServiceAccount"
        "name" = "mpi-operator"
        "namespace" = "mpi-operator"
      },
    ]
  }
}
```
```hcl
resource "kubernetes_manifest" "customresourcedefinition_mpijobs_kubeflow_org" {
  provider = kubernetes-alpha
  manifest = {
    "apiVersion" = "apiextensions.k8s.io/v1beta1"
    "kind" = "CustomResourceDefinition"
    "metadata" = {
      "name" = "mpijobs.kubeflow.org"
      "namespace" = "default"
    }
    "spec" = {
      "additionalPrinterColumns" = [
        {
          "JSONPath" = ".metadata.creationTimestamp"
          "name" = "Age"
          "type" = "date"
        },
      ]
      "group" = "kubeflow.org"
      "names" = {
        "kind" = "MPIJob"
        "plural" = "mpijobs"
        "shortNames" = [
          "mj",
          "mpij",
        ]
        "singular" = "mpijob"
      }
      "scope" = "Namespaced"
      "subresources" = {
        "status" = {}
      }
      "validation" = {
        "openAPIV3Schema" = {
          "properties" = {
            "spec" = {
              "properties" = {
                "mpiReplicaSpecs" = {
                  "properties" = {
                    "Launcher" = {
                      "properties" = {
                        "replicas" = {
                          "maximum" = 1
                          "minimum" = 1
                          "type" = "integer"
                        }
                      }
                    }
                    "Worker" = {
                      "properties" = {
                        "replicas" = {
                          "minimum" = 1
                          "type" = "integer"
                        }
                      }
                    }
                  }
                }
                "slotsPerWorker" = {
                  "minimum" = 1
                  "type" = "integer"
                }
              }
            }
          }
        }
      }
      "version" = "v1"
    }
  }
}

resource "kubernetes_manifest" "deployment_mpi_operator" {
  provider = kubernetes-alpha
  manifest = {
    "apiVersion" = "apps/v1"
    "kind" = "Deployment"
    "metadata" = {
      "labels" = {
        "app" = "mpi-operator"
      }
      "name" = "mpi-operator"
      "namespace" = "mpi-operator"
    }
    "spec" = {
      "replicas" = 1
      "selector" = {
        "matchLabels" = {
          "app" = "mpi-operator"
        }
      }
      "template" = {
        "metadata" = {
          "labels" = {
            "app" = "mpi-operator"
          }
        }
        "spec" = {
          "containers" = [
            {
              "args" = [
                "-alsologtostderr",
                "--kubectl-delivery-image",
                "mpioperator/kubectl-delivery:latest",
              ]
              "command" = [
                "/opt/mpi-operator.v1",
              ]
              "image" = "mpioperator/mpi-operator:latest"
              "imagePullPolicy" = "Always"
              "name" = "mpi-operator"
            },
          ]
          "serviceAccountName" = "mpi-operator"
        }
      }
    }
  }
}
```
The mpijob I'm using comes from [mpi-operator tensorflow-benchmarks example](https://github.com/kubeflow/mpi-operator/blob/master/examples/v1/tensorflow-benchmarks.yaml) and the converted version is given below
```hcl
resource "kubernetes_manifest" "mpijob_tensorflow_benchmarks" {
  provider = kubernetes-alpha
  manifest = {
    "apiVersion" = "kubeflow.org/v1"
    "kind" = "MPIJob"
    "metadata" = {
      "name" = "tensorflow-benchmarks"
      "namespace" = "default"
    }
    "spec" = {
      "cleanPodPolicy" = "Running"
      "mpiReplicaSpecs" = {
        "Launcher" = {
          "replicas" = 1
          "template" = {
            "spec" = {
              "containers" = [
                {
                  "command" = [
                    "mpirun",
                    "--allow-run-as-root",
                    "-np",
                    "2",
                    "-bind-to",
                    "none",
                    "-map-by",
                    "slot",
                    "-x",
                    "NCCL_DEBUG=INFO",
                    "-x",
                    "LD_LIBRARY_PATH",
                    "-x",
                    "PATH",
                    "-mca",
                    "pml",
                    "ob1",
                    "-mca",
                    "btl",
                    "^openib",
                    "python",
                    "scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py",
                    "--model=resnet101",
                    "--batch_size=64",
                    "--variable_update=horovod",
                  ]
                  "image" = "mpioperator/tensorflow-benchmarks:latest"
                  "name" = "tensorflow-benchmarks"
                },
              ]
            }
          }
        }
        "Worker" = {
          "replicas" = 2
          "template" = {
            "spec" = {
              "containers" = [
                {
                  "image" = "mpioperator/tensorflow-benchmarks:latest"
                  "name" = "tensorflow-benchmarks"
                  "resources" = {
                    "limits" = {
                      "nvidia.com/gpu" = 1
                    }
                  }
                },
              ]
            }
          }
        }
      }
      "slotsPerWorker" = 1
    }
  }
}
```

### Panic Output

[crash.log](https://gist.github.com/jefflantz/ef5ae5a8627f424542418948b4a5eb0f)

### Expected Behavior
An mpijob resource should be created, (as well as a namespace, service account, cluster role, cluster role binding, custom resource definition, and deployment)

### Actual Behavior
Terraform crashes on the last step: creating the mpijob.

### Steps to Reproduce
Let the blocks of hcl code correspond to the following files, respectively `namespace.tf`, `staging/mpi-operator.tf`, `staging/mpijob.tf`, `staging/tensorflow-benchmarks.tf` and run
```bash
terraform init
terraform apply --auto-approve
cp staging/mpi-operator.tf .
terraform apply --auto-approve
cp staging/mpijob.tf .
terraform apply --auto-approve
cp staging/tensorflow-benchmarks.tf .
terraform apply --auto-approve
```

### Important Factoids
I'm running locally on minikube. I don't have gpus, but that error would occur after the mpijob is created (as I have tested on a GKE cluster). The [.yaml file](https://github.com/kubeflow/mpi-operator/blob/master/examples/v1/tensorflow-benchmarks.yaml) that was translated into tensorflow-benchmarks.tf works just fine by running `kubectl create -f tensorflow-benchmarks.yaml` after creating all other resources with terraform.

I've tried this with a different file that runs the HPL benchmarks from a different project I'm working on, and that gives what appears to be the same error.

Notably, this crd uses `apiextensions.k8s.io/v1beta1`, which is somewhat older and possibly less supported.

### References
The staging is required due to PR-41 being a WIP

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by alexsomesan** [comment]
Unfortunately, this is a known issue in Terraform itself. According to my tests it's been fixed in TF 0.13.
Can you try the same scenario using Terraform v0.13.0 RC1? https://github.com/hashicorp/terraform/releases/tag/v0.13.0-rc1
It'd be great to validate if it's indeed resolved there.

**Comment by jefflantz** [comment]
I downloaded Terraform v0.13.0 RC1 and (eventually) figured out how to compile it with the using script/build.sh and cp'ing it out of the created docker container. Some extra documentation on that might be nice, but oh well. `terraform 0.13upgrade` worked reasonably well, but it took a while to figure out where to put the `terraform-provider-kubernetes-alpha` binary. I ended up going with `./.terraform/plugins/hashicorp.com/kubernetes/kubernetes-alpha/0.1.0/linux_amd64/` which worked. All that fixed the crash, and then I needed to resolve a namespace issue and it works now! Many thanks and closed :D

**Comment by alexsomesan** [comment]
Sorry for not mentioning this earlier. I use this tool to manage Terraform versions: https://github.com/asdf-vm/asdf
It makes it dead easy to switch between them (and other tools) with a single cli command.

Glad to hear you issue is resolved by 0.13.
